test/e2e_clutz_dts_test.ts

```diff
@@ -0,0 +1,89 @@
+
+import {expect} from 'chai';
+import * as ts from 'typescript';
+
+import * as tsickle from '../src/tsickle';
+
+import {compileWithTransfromer, createProgram, dtsCompilerOptions} from './test_support';
+
+describe('clutz dts', () => {
+  it('does nothing if there are no exports', () => {
+    const sources = new Map<string, string>();
+    sources.set('input.ts', `const x = 1;`);
+
+    const {files} = compileWithTransfromer(sources, dtsCompilerOptions);
+    const dts = files.get('input.d.ts')!;
+
+    expect(dts).not.to.contain('namespace ಠ_ಠ.clutz');
+  });
+
+  it('produces a valid .d.ts', () => {
+    const sources = new Map<string, string>();
+    sources.set('input.ts', `export const x = 1;`);
+
+    const {files} = compileWithTransfromer(sources, dtsCompilerOptions);
+    const dts = files.get('input.d.ts')!;
+
+    const program = createProgram(new Map([['input.d.ts', dts]]));
+    const diagnostics = ts.getPreEmitDiagnostics(program);
+
+    expect(diagnostics, tsickle.formatDiagnostics(diagnostics)).to.be.empty;
+  });
+
+  it('reexports things in the clutz namespace', () => {
+    const sources = new Map<string, string>();
+    sources.set('input.ts', `
+      export const x = 1;
+      export function incr(n) { return x + 1; }`);
+
+    const {files} = compileWithTransfromer(sources, dtsCompilerOptions);
+    const dts = files.get('input.d.ts')!;
+
+    expect(dts).to.contain('namespace ಠ_ಠ.clutz');
+    expect(dts).to.contain('const module$contents$input_x: typeof x;');
+    expect(dts).to.contain('const module$contents$input_incr: typeof incr;');
+  });
+
+  it('exports the type and value of a class', () => {
+    const sources = new Map<string, string>();
+    sources.set('input.ts', `export class foo {}`);
+
+    const {files} = compileWithTransfromer(sources, dtsCompilerOptions);
+    const dts = files.get('input.d.ts')!;
+
+    expect(dts).to.contain('type module$contents$input_foo = foo;');
+    expect(dts).to.contain('const module$contents$input_foo: typeof foo;');
+  });
+
+  it('handles multisegment paths', () => {
+    const sources = new Map<string, string>();
+    sources.set('path/to/file/input.ts', `export const x = 1;`);
+
+    const {files} = compileWithTransfromer(sources, dtsCompilerOptions);
+    const dts = files.get('path/to/file/input.d.ts')!;
+
+    expect(dts).to.contain('const module$contents$path$to$file$input_x: typeof x;');
+  });
+
+  it(`doesn't error on es5 style exports`, () => {
+    const sources = new Map<string, string>();
+    sources.set('path/to/file/input.ts', `let x = 0; export = x;`);
+
+    const {files} = compileWithTransfromer(sources, dtsCompilerOptions);
+    const dts = files.get('path/to/file/input.d.ts')!;
+
+    const program = createProgram(new Map([['input.d.ts', dts]]));
+    const diagnostics = ts.getPreEmitDiagnostics(program);
+
+    expect(diagnostics, tsickle.formatDiagnostics(diagnostics)).to.be.empty;
+  });
```

Comment:
You only have tests exporting values. Can you add tests that export types, e.g. `export interface ...`, `export type ...`?

Reply:
Done.